File: ios-client/AWSCognito/AWSCognito/Data/Repositories/Auth/AuthEndpoints.swift
//
//  AuthEndpoints.swift
//  AWSCognito
//

import Foundation

enum AuthEndpoints {

    struct ExchangeGoogleToken: Endpoint {
        var url: String
        var method: HTTPMethod = .post
        var body: (any Encodable)?

        init(idToken: String, email: String?, fullName: String?) {
            self.url = "\(Config.value(for: .api))/auth/google"
            self.body = GoogleAuthRequest(
                idToken: idToken,
                email: email,
                fullName: fullName
            )
        }
    }

    struct ExchangeAppleToken: Endpoint {
        var url: String
        var method: HTTPMethod = .post
        var body: (any Encodable)?

        init(identityToken: String, authorizationCode: String, email: String?, fullName: String?) {
            self.url = "\(Config.value(for: .api))/auth/apple"
            self.body = AppleAuthRequest(
                identityToken: identityToken,
                authorizationCode: authorizationCode,
                email: email,
                fullName: fullName
            )
        }
    }
}

File: android-client/app/src/main/java/com/example/awscognito/data/repositories/auth/AuthEndpoints.kt
package com.example.awscognito.data.repositories.auth

import com.example.awscognito.data.model.AppleAuthRequest
import com.example.awscognito.data.model.AuthTokenResponse
import com.example.awscognito.data.model.GoogleAuthRequest
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthEndpoints {

    @POST("auth/google")
    suspend fun exchangeGoogleToken(
        @Body request: GoogleAuthRequest
    ): AuthTokenResponse

    @POST("auth/apple")
    suspend fun exchangeAppleToken(
        @Body request: AppleAuthRequest
    ): AuthTokenResponse
}